libc/src/stdio/scanf_core/parser.h

```diff
@@ -94,6 +102,203 @@ class Parser {
 #endif // LIBC_COPT_SCANF_DISABLE_INDEX_MODE
 };
 
+template <typename ArgProvider>
```

Comment:
same as printf, I don't think these functions need to be templated individually.